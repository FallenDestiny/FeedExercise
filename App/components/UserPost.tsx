import React, { ReactElement } from 'react'
import { TouchableOpacity, View, Image, Text } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import Divider from './Divider'
import { PostData } from '../redux/App/actions'
import fonts from '../assets/fonts'
import { colors } from '../utils/colors'
var moment = require('moment');

interface Props {
    post: PostData,
}

export default function UserPost({ post }: Props): ReactElement {
    return (
        <View style={styles.holder}>
            <View style={styles.mainContainer}>
                {
                    post && post.image ?
                        <Image style={styles.image} source={{ uri: post.image }} /> : null
                }

                {
                    post && post.title && post.createdAt ?
                        <View style={styles.detailsInfo}>
                            <Text style={styles.title}>{post.title}</Text>
                            <Text style={styles.created}>{moment(post.createdAt).fromNow()}</Text>
                        </View> : null
                }

            </View>
            <Divider style={styles.divider} />
        </View>
    )
}

const styles = EStyleSheet.create({
    holder: {
        flex: 1,
        width: '100%',
        height: '100rem',
        marginTop: '30rem',
    },
    mainContainer: {
        flex: 1,
        marginBottom: '16rem',
        flexDirection: 'row'
    },
    divider: {
        width: '100%'
    },
    image: {
        aspectRatio: 1,
        width: undefined,
        height: '100%',
        resizeMode: 'cover',
        borderRadius: '4rem'
    },
    detailsInfo: {
        flex: 1,
        marginLeft: '15rem',
        marginRight: '8rem'
    },
    title: {
        fontFamily: fonts.sf_pro_display_semibold,
        fontSize: '15rem',
        color: colors.color1,
    },
    created: {
        fontFamily: fonts.sf_pro_display_medium,
        fontSize: '12rem',
        color: colors.color2,
    }
})
