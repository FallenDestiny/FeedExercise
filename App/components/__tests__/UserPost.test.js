import React from 'react';
import renderer from 'react-test-renderer';
import UserPost from '../UserPost'

const post =
{
    "id": "1",
    "createdAt": "2020-02-18T04:40:49.601Z",
    "title": "Principal Identity Associate",
    "image": "https://images.unsplash.com/photo-1582043725042-f3d1873eeadf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
    "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
    "likes": 68,
    "comments": 43
}


describe('UserHeader', () => {
    describe('Rendering', () => {
        it('draw all the user post with data', () => {

            const component = renderer.create(<UserPost post={post}/>).toJSON()
            expect(component).toMatchSnapshot()
        });


        it('draw all the user post with image only', () => {

            const component = renderer.create(<UserPost post={{image:post.image}}/>).toJSON()
            expect(component).toMatchSnapshot()
        });

        it('draw all the user post with description and createDate', () => {

            const component = renderer.create(<UserPost post={{description: post.description, createdAt:post.createdAt}}/>).toJSON()
            expect(component).toMatchSnapshot()
        });

        it('draw an empty header', () => {
            const component = renderer.create(<UserPost />).toJSON()
            expect(component).toMatchSnapshot()
        });
    });

})
