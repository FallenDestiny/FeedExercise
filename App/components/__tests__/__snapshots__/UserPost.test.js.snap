// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHeader Rendering draw all the user post with data 1`] = `
<View>
  <View>
    <Image
      source={
        Object {
          "uri": "https://images.unsplash.com/photo-1582043725042-f3d1873eeadf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
        }
      }
    />
    <View>
      <Text>
        Principal Identity Associate
      </Text>
      <Text>
        9 days ago
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`UserHeader Rendering draw all the user post with description and createDate 1`] = `
<View>
  <View />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`UserHeader Rendering draw all the user post with image only 1`] = `
<View>
  <View>
    <Image
      source={
        Object {
          "uri": "https://images.unsplash.com/photo-1582043725042-f3d1873eeadf?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`UserHeader Rendering draw an empty header 1`] = `
<View>
  <View />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E4E4E499",
          "height": 1,
          "width": "80%",
        },
        undefined,
      ]
    }
  />
</View>
`;
